src/helper/audio/audio.cpp

```diff
@@ -338,18 +343,27 @@ namespace Soundux::Objects
             Globals::gAudio.onSoundProgressed(sound, readFrames);
         }
 
-        if (readFrames <= 0)
+        if (sound->repeat && !sound->shouldSeek && readFrames < frameCount)
         {
-            if (sound->repeat)
+            while (frameCount > 0 && readFrames > 0)
             {
+                frameCount -= readFrames;
                 ma_decoder_seek_to_pcm_frame(sound->raw.decoder, 0);
-                Globals::gAudio.onSoundSeeked(sound, 0);
-            }
-            else
-            {
-                Globals::gQueue.push_unique(reinterpret_cast<std::uintptr_t>(device),
-                                            [sound = *sound] { Globals::gAudio.onFinished(sound); });
+                output = reinterpret_cast<void *>(
+                    reinterpret_cast<ma_uint8 *>(output) +
+                    (readFrames)*ma_get_bytes_per_frame(sound->raw.decoder.load()->outputFormat,
```

Comment:
Just a minor thing but could you remove the brackets around readFrames?

Reply:
Done